homeassistant/components/smarty/sensor.py

```diff
@@ -171,9 +171,12 @@ def __init__(self, name, smarty):
                          device_class=DEVICE_CLASS_TIMESTAMP,
                          unit_of_measurement=None,
                          smarty=smarty)
+        self._days_left = 90
 
     def update(self) -> None:
         """Update state."""
         _LOGGER.debug('Updating sensor %s', self._name)
         days_left = self._smarty.filter_timer
-        self._state = dt_util.now() + dt.timedelta(days=days_left)
+        if days_left < self._days_left:
+            self._state = dt_util.now() + dt.timedelta(days=days_left)
```

Comment:
Won't the time part of `dt_util.now` change on each update? We should set the time part to some fixed time so that the state doesn't change unless days_left changes.

Reply:
I think it will not. _smarty.filter_timer gives a number between 0-90. On each update the _state will change only if _smarty.filter_timer changes to 89 for example and the _state will be dt_util.now() + 89 days. Maybe self._days_left should be defaulted to 91 so the state to be updated on first update.

Reply:
Right. Yeah it's good to make sure we get a state in first update. :+1: